### PR TITLE
Add Specular property to DirectionalLight3D

### LINKED ISSUE #80737: DirectionalLight3D currently has no Specular property
_Note: This issue was transferred from the godot_docs repository. What appeared to be an issue with the documentation turned out to be an issue with the feature itself. clayjohn has commented below a bit of background. Ultimately, the ghost documentation highlights that we removed a setting by mistake. The solution is to restore the setting._

**Your Godot version:** 4.1 Stable

**Issue description:** DirectionalLight3D missing Specular property, current documentation states this should exist

**URL to the documentation page:**
https://docs.godotengine.org/en/stable/tutorials/3d/lights_and_shadows.html
https://docs.godotengine.org/en/stable/tutorials/3d/global_illumination/faking_global_illumination.html

**Additional information:**

https://github.com/godotengine/godot-docs/blob/05fe756d9d1e6abab198da4721ba8f0407f0e45e/tutorials/3d/lights_and_shadows.rst?plain=1#L44

According to the text (from Godot Engine 4.1 - 3D Lights and Shadows, linked above) and the screenshot of common light properties above it, `DirectionalLight3D` (like Omni and Spot lights) should have a `Specular` property but currently doesn't (4.1 Stable). Not sure if this is a bug, a documentation issue or due to on going work.

directionallight3d_missingspecular
<sub>specular property missing from DirectionalLight3D</sub>

Due to this, currently unable to remove the specular blob from DirectionalLights when following "Godot Engine 4.1 - Faking global illumination". Which specifically mentions setting `Specular: 0.0` on `DirectionalLight3D`.

https://github.com/godotengine/godot-docs/blob/05fe756d9d1e6abab198da4721ba8f0407f0e45e/tutorials/3d/global_illumination/faking_global_illumination.rst?plain=1#L48

### PR DESCRIPTION
**Pull request by AndrewShobbrook**
I've re-added the specular property for the DirectionalLight3D to fix #80737.

### CODE REVIEW

**Comment by AThousandShips** [comment]
Thank you for opening this PR!

The documentation needs to be updated to reflect the new default values, see the test with a cross next to it

**Comment by AndrewShobbrook** [comment]
> Thank you for opening this PR!
> 
> The documentation needs to be updated to reflect the new default values, see the test with a cross next to it

I think I've fixed it, but I'm not entirely sure whether or not the documentation is correct

**Review by Calinou — COMMENTED** [review]
Tested locally, it works as expected, including in situations with more than one DirectionalLight visible.

However, the Light Specular property appears twice in the inspector (look at the bottom of the inspector for the "correct" one):

image

File: scene/3d/light_3d.cpp [hunk 1]
```diff
@@ -567,6 +567,7 @@ void DirectionalLight3D::_bind_methods() {
 	ADD_PROPERTYI(PropertyInfo(Variant::FLOAT, "directional_shadow_fade_start", PROPERTY_HINT_RANGE, "0,1,0.001"), "set_param", "get_param", PARAM_SHADOW_FADE_START);
 	ADD_PROPERTYI(PropertyInfo(Variant::FLOAT, "directional_shadow_max_distance", PROPERTY_HINT_RANGE, "0,8192,0.1,or_greater,exp"), "set_param", "get_param", PARAM_SHADOW_MAX_DISTANCE);
 	ADD_PROPERTYI(PropertyInfo(Variant::FLOAT, "directional_shadow_pancake_size", PROPERTY_HINT_RANGE, "0,1024,0.1,or_greater,exp"), "set_param", "get_param", PARAM_SHADOW_PANCAKE_SIZE);
+	ADD_PROPERTYI(PropertyInfo(Variant::FLOAT, "light_specular", PROPERTY_HINT_RANGE, "0,16,0.001,or_greater"), "set_param", "get_param", PARAM_SPECULAR);
```
Comment (Calinou) [inline scene/3d/light_3d.cpp:570]: The fix for the issue I mentioned is likely to remove that line:

```suggestion
```

File: doc/classes/DirectionalLight3D.xml [hunk 2]
```diff
@@ -35,6 +35,8 @@
 		<member name="directional_shadow_split_3" type="float" setter="set_param" getter="get_param" default="0.5">
 			The distance from shadow split 2 to split 3. Relative to [member directional_shadow_max_distance]. Only used when [member directional_shadow_mode] is [constant SHADOW_PARALLEL_4_SPLITS].
 		</member>
+		<member name="light_specular" type="float" setter="set_param" getter="get_param" default="1.0">
```
Comment (AThousandShips) [inline doc/classes/DirectionalLight3D.xml:38]: Remove this, the property is inherited and has the same default

**Pull request closed without merging**